src/arche/arche.py

```diff
@@ -140,7 +140,7 @@ def run_all_rules(self):
         self.run_schema_rules()
 
     def data_quality_report(self, bucket: Optional[str] = None):
-        if helpers.is_collection_key(self.source):
+        if helpers.is_collection_key(str(self.source or "")):
```

Comment:
I don't think self.source could be `None` or False, so `str(self.source)` should be enough